### ISSUE TITLE
Envoy reports deprecation

### ISSUE THREAD

**Issue by sbezverk** [istio/istio#11960]
**Describe the bug**
pilot log filled with this warning:
```
[2019-02-21 15:21:07.272][15][warning][misc] [external/envoy/source/common/protobuf/utility.cc:129] Using deprecated option 'envoy.api.v2.listener.Filter.config'. This configuration will be removed from Envoy soon. Please see https://github.com/envoyproxy/envoy/blob/master/DEPRECATED.md for details.
```

It is experienced with release-1.1 branch

**Comment by kyessenov** [comment]
You need to set ISTIO_PROXY_VERSION to 1.1 or later for this to go away.

**Comment by kyessenov** [comment]
Is this happening per request or per config load?

**Comment by sbezverk** [comment]
@kyessenov it looks like a series of message per config load:
```
[2019-02-21 15:21:07.272][15][warning][misc] [external/envoy/source/common/protobuf/utility.cc:129] Using deprecated option 'envoy.api.v2.listener.Filter.config'. This configuration will be removed from Envoy soon. Please see https://github.com/envoyproxy/envoy/blob/master/DEPRECATED.md for details.
[2019-02-21 15:21:07.272][15][warning][misc] [external/envoy/source/common/protobuf/utility.cc:129] Using deprecated option 'envoy.api.v2.listener.Filter.config'. This configuration will be removed from Envoy soon. Please see https://github.com/envoyproxy/envoy/blob/master/DEPRECATED.md for details.
[2019-02-21 15:21:07.272][15][warning][misc] [external/envoy/source/common/protobuf/utility.cc:129] Using deprecated option 'envoy.api.v2.listener.Filter.config'. This configuration will be removed from Envoy soon. Please see https://github.com/envoyproxy/envoy/blob/master/DEPRECATED.md for details.
              - decorator:
[2019-02-21 15:21:07.272][15][warning][misc] [external/envoy/source/common/protobuf/utility.cc:129] Using deprecated option 'envoy.api.v2.listener.Filter.config'. This configuration will be removed from Envoy soon. Please see https://github.com/envoyproxy/envoy/blob/master/DEPRECATED.md for details.
[2019-02-21 15:21:07.272][15][warning][misc] [external/envoy/source/common/protobuf/utility.cc:129] Using deprecated option 'envoy.api.v2.Cluster.hosts'. This configuration will be removed from Envoy soon. Please see https://github.com/envoyproxy/envoy/blob/master/DEPRECATED.md for details.
[2019-02-21 15:21:07.272][15][warning][misc] [external/envoy/source/common/protobuf/utility.cc:129] Using deprecated option 'envoy.api.v2.Cluster.hosts'. This configuration will be removed from Envoy soon. Please see https://github.com/envoyproxy/envoy/blob/master/DEPRECATED.md for details.
[2019-02-21 15:21:07.272][15][warning][misc] [external/envoy/source/common/protobuf/utility.cc:129] Using deprecated option 'envoy.api.v2.Cluster.hosts'. This configuration will be removed from Envoy soon. Please see https://github.com/envoyproxy/envoy/blob/master/DEPRECATED.md for details.
[2019-02-21 15:21:07.279][15][warning][misc] [external/envoy/source/common/protobuf/utility.cc:129] Using deprecated option 'envoy.config.filter.network.http_connection_manager.v2.HttpFilter.config'. This configuration will be removed from Envoy soon. Please see https://github.com/envoyproxy/envoy/blob/master/DEPRECATED.md for details.
[2019-02-21 15:21:07.279][15][warning][misc] [external/envoy/source/common/protobuf/utility.cc:129] Using deprecated option 'envoy.config.filter.accesslog.v2.AccessLog.config'. This configuration will be removed from Envoy soon. Please see https://github.com/envoyproxy/envoy/blob/master/DEPRECATED.md for details.
```

**Comment by kyessenov** [comment]
OK. I think we can tolerate extra noise for 1.1. We can patch the static configs after the release to use `typed_configs`. Static configs are only loaded once per proxy restart AFAICT.

**Comment by rvennam** [comment]
I'm seeing similar log data in the proxy logs for my application.
`
[2019-04-02 15:39:00.087][16][warning][misc] [external/envoy/source/common/protobuf/utility.cc:129] Using deprecated option'envoy.api.v2.route.Route.per_filter_config'. This configuration will be removed from Envoy soon. Please see https://github.com/envoyproxy/envoy/blob/master/DEPRECATED.md for details
`

**Comment by esnible** [comment]
This is a duplicate of https://github.com/istio/istio/issues/11960 .  I don't know which Issue should be authoritative and which should be closed.

**Comment by howardjohn** [comment]
This is fixed in 1.2, see #2234

As far as actually migrating off the deprecated features, I believe there is work being done in these areas. if not we should open issues for those, but the actual log issue has been solved
